service_thing_aggregate/src/main/scala/controllers/ObjectSearchController.scala

```diff
@@ -0,0 +1,49 @@
+package controllers
+
+import com.google.inject.Inject
+import models.MusitThing
+import no.uio.musit.service.MusitResults.{MusitDbError, MusitError, MusitResult, MusitSuccess}
+import play.api.Logger
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.{Action, Controller, Result, Results}
+import services.ObjectSearchService
+
+/**
+ * Created by jarle on 05.10.16.
+ */
+
+object Utils {
+
+  //TODO: Move this to somewhere common
+  def musitResultToPlayResult[A](musitResult: MusitResult[A], toJsonTransformer: A => JsValue): Result = {
+    musitResult match {
+      case MusitSuccess(res) =>
+        Results.Ok(toJsonTransformer(res))
+
+      case MusitDbError(msg, ex) =>
+        Logger.error(msg, ex.orNull)
+        Results.InternalServerError(msg)
+
+      case r: MusitError =>
+        Results.InternalServerError(r.message)
+    }
+  }
+}
+
+class ObjectSearchController @Inject() (
+    service: ObjectSearchService
+) extends Controller {
+
+  private val maxLimit = 100
+
+  def search(mid: Int, museumNo: String = "", subNo: String = "", term: String = "", page: Int = 1, limit: Int = 25) =
+    Action.async { request =>
+
+      val limitToUse = Math.max(limit, maxLimit)
```

Comment:
No, that shouldn't be necessary. What you can do to make this easier to test, is to make the `maxLimit` property configurable through the `application.conf` file in `service_thing_aggregate`. That way you can set the property to a much lower value during testing to avoid loading 100s of rows of data.

In `application.conf` you can add the following property at the end of the file:
```
musit.thing.search.max-limit = 100
```
When setting up the spec (for e.g. the integration testing), you can add a config file in `src/it/resources/application.conf` with the same config parameter, but with a lower value. When starting the tests, this value will be used instead of the one defined in `src/main/resources/application.conf`. (Same applies in `src/test/resources/application.conf`)